crates/settlement-clients/ethereum/src/tests/mod.rs

```diff
@@ -0,0 +1,318 @@
+use color_eyre::eyre::eyre;
+use rstest::*;
+use settlement_client_interface::SettlementVerificationStatus;
+use std::env;
+use std::io::BufRead;
+use std::path::PathBuf;
+use std::time::Duration;
+use std::{
+    fs::{self, File},
+    io::BufReader,
+    str::FromStr,
+};
+use tokio::time::sleep;
+use utils::env_utils::get_env_var_or_panic;
+
+use settlement_client_interface::SettlementClient;
+use utils::settings::default::DefaultSettingsProvider;
+
+use crate::conversion::to_padded_hex;
+use crate::EthereumSettlementClient;
+use alloy::providers::Provider;
+use alloy_primitives::FixedBytes;
+
+// Using the Pipe trait to write chained operations easier
+trait Pipe: Sized {
+    fn pipe<T, F: FnOnce(Self) -> T>(self, f: F) -> T {
+        f(self)
+    }
+}
+
+// Implement Pipe for all types
+impl<S> Pipe for S {}
+
+// TODO: betterment of file routes
+
+use lazy_static::lazy_static;
+
+lazy_static! {
+    static ref ENV_FILE_PATH: PathBuf = PathBuf::from(".env.test");
+    static ref CURRENT_PATH: String = env::current_dir()
+        .expect("Failed to get current directory")
+        .to_str()
+        .expect("Path contains invalid Unicode")
+        .to_string();
+    static ref PORT: u16 = 3000_u16;
+    static ref ETH_RPC: String = "https://eth.llamarpc.com".to_string();
+    static ref SHOULD_IMPERSONATE_ACCOUNT: bool = get_env_var_or_panic("TEST_IMPERSONATE_OPERATOR") == *"1";
+    static ref TEST_DUMMY_CONTRACT_ADDRESS: String = get_env_var_or_panic("TEST_DUMMY_CONTRACT_ADDRESS");
+    static ref STARKNET_OPERATOR_ADDRESS: Address =
+        Address::from_str("0x2C169DFe5fBbA12957Bdd0Ba47d9CEDbFE260CA7").expect("Could not impersonate account.");
+    static ref STARKNET_CORE_CONTRACT_ADDRESS: Address =
+        Address::from_str("0xc662c410c0ecf747543f5ba90660f6abebd9c8c4").expect("Could not impersonate account.");
+}
+
+pub struct TestFixture {
+    pub ethereum_settlement_client: EthereumSettlementClient,
+    pub provider: alloy::providers::RootProvider<alloy::transports::http::Http<reqwest::Client>>,
+}
+
+fn ethereum_test_fixture(block_no: u64) -> TestFixture {
+    // Load ENV vars
+    dotenvy::from_filename(&*ENV_FILE_PATH).expect("Could not load .env.test file.");
+
+    // Setup Anvil
+    let anvil = Anvil::new()
+        .port(*PORT)
+        .fork(&*ETH_RPC)
+        .fork_block_number(block_no - 1)
+        .try_spawn()
+        .expect("Could not spawn Anvil.");
+
+    // Setup Provider
+    let provider = ProviderBuilder::new().on_http(anvil.endpoint_url());
+
+    // Setup EthereumSettlementClient
+    let settings_provider: DefaultSettingsProvider = DefaultSettingsProvider {};
+    let ethereum_settlement_client = EthereumSettlementClient::with_test_settings(&settings_provider, provider.clone());
+
+    TestFixture { ethereum_settlement_client, provider }
+}
+
+#[rstest]
+#[tokio::test]
+#[case::basic(20468828)]
+async fn update_state_blob_with_dummy_contract_works(#[case] block_no: u64) {
```

Comment:
can you also explain the rationale here for having the two test cases

Reply:
Each function tests different code flow of `update_state_blob` method, 
1) `with_dummy_contract` : tests if the method is able to do a transaction with same function selector on a dummy contract.
2) `with_impersonation` : tests if the method is able to impersonate the`Starknet Operator` and do an `update_state` transaction.

Added as docs for tests

Reply:
I see, I actually meant the rationale we had initially that when we impersonate we are not checking the signing logic and that's why we wrote this test

Reply:
valid, Added.